### PR TITLE
calculate distance matrices to sample weighted by distance bands

### PR DESCRIPTION
**Pull request by gboeing**
We do not need to merge this PR yet... I'm just opening it now for a place to discuss/comment.

### CODE REVIEW

**Comment by gboeing** [comment]
@smmaurer I'll benchmark number 1 later this week. I wrote the great circle routine myself, and it's all vectors, so it's fast. I'm using a scipy routine for the euclidean distance matrix, and I'm not sure how they implemented it. If their routine isn't particularly fast, it'll be easy enough to re-write it myself fully vectorized with numpy.

Regarding number 2, I liked the hierarchical index because it seems more pandas-like, makes for fast indexing, and matches the (presumed) structure we'd want to use in a relational DB (i.e., columns of origin_key, destination_key and distance -- with indices for the two keys). We can inspect the memory requirements of each format as part of the decision of how to represent the data. That said, it's trivial to unstack the dataframe into an *i* x *j* matrix instead of a stacked, multi-indexed vector.

I'll check out the above and lastly will add some more tests of the distance matrix module before doing a merge (probably later this week).

File: data/README.md [hunk 1]
```diff
@@ -4,4 +4,6 @@ Information about the survey: http://www.dot.ca.gov/hq/tpp/offices/omsp/statewid
 
 Data download: https://www.nrel.gov/transportation/secure-transportation-data.html
 
-The data is open access, but you will need to fill out a registration form. Download the file named `caltrans_full_survey.zip`, 233.2 MB, and place it in this directory. 
\ No newline at end of file
+The data is open access, but you will need to fill out a registration form. Download the file named `caltrans_full_survey.zip`, 233.2 MB, and place it in this directory.
+
+You can download the California census tracts shapefile `tl_2010_06_tract10` from: https://www.census.gov/cgi-bin/geo/shapefiles/index.php?year=2016&layergroup=Census+Tracts
```
Comment (smmaurer) [inline data/README.md]: Wrong url, I think -- should be "year=2010"

**Comment by smmaurer** [comment]
@gboeing Ok, great! I think we're on the same page about number 2 -- I agree that a hierarchical index seems better. The distance matrix demo notebook currently generates an i x j CSV, but we can change it to [i x j] x 3, with columns for the origin key, destination key, and distance.

Thanks again for putting together these distance calculations! Later this week I'll take a look at integrating them into the merging/sampling code..

**Comment by gboeing** [comment]
@smmaurer regarding benchmarking euclidean vs great circle calculations... the wall time in the notebook was actually 34 ms (euclidean) vs 1.47 s (great circle). So great circle takes about 43x longer to run. So, the scipy routine seems sufficiently fast and I'll leave that as is.

File: data/README.md (hunk 1 above)
Comment (gboeing) [inline data/README.md]: fixed by 9bf6452

**Comment by smmaurer** [comment]
@gboeing Oh, that makes more sense. Sorry for misreading it!

**Pull request merged**